manifests/agent.pp

```diff
@@ -53,8 +53,8 @@
 
   # Server configuration
 
-  $ossec_ip                          = $wazuh::params_agent::ossec_ip,
-  $ossec_hostname                    = $wazuh::params_agent::ossec_hostname,
+  $ossec_registration_ip             = $wazuh::params_agent::ossec_registration_ip,
```

Comment:
Why are we postfixing the parameter name with`_ip`? Wazuh supports both an IP or a domain name.

We could name those parameters `wazuh_master_endpoint` and `wazuh_workers_enpoint`.

Reply:
You are right! Domains can also be specified and this naming could confuse users.

I like the naming you propose for cluster but I think it could be confusing for non-cluster configurations, as you would have to declare master and worker endpoint for same IP/domain and user that are not familiar with cluster could be confused in my opinion.

Something like: `wazuh_register_endpoint` and `wazuh_reporting_endpoint` comes to my mind in a quick approach

Reply:
We don't need to declare both `wazuh_master_endpoint` and `wazuh_workers_enpoint` if we bring back double `<address>` in the ossec.conf file:
```
  <%- if @wazuh_register_endpoint then -%>	
    <address><%= @wazuh_master_endpoint %></address>	
  <%- end -%>
  <%- if @wazuh_reporting_endpoint then -%>	
    <address><%= @wazuh_workers_enpoint %></address>	
  <%- end -%>
```

The Wazuh agent takes the last one as I wrote in https://github.com/wazuh/wazuh-puppet/issues/110#issuecomment-479480063.

But it conflicts with my suggestion of setting `127.0.0.1` as a default value for both 😄 If a user of the Puppet modules sets a value for `wazuh_master_endpoint`, but not for `wazuh_workers_endpoint`, he would end up with his agents reporting to `127.0.0.1`.

I agree with your naming suggestion and I think setting `127.0.0.1` as a default value might cause issues, so lets keep undef! 🎉

Reply:
That's exactly what I tried to point out, I think this way users have more flexibility and they could even register agents without configuring them to do it later or in with other tool.

Anyway, we are really happy about this kind of discussion since it helps us to understand the user's needs and make the code better.

Feel free to comment anytime, feedback is always welcome!